hub/scripts.js

```diff
@@ -397,6 +397,39 @@
     });
   }
 
+  async function decoratePromotions() {
+    if (document.querySelector('main .promotion') instanceof HTMLElement) {
+      // A promotion has already been defined on the page.
+      // Do not inject another one.
+      return;
+    }
+
+    let promotionName = document.querySelector('head meta[name="promotion"]');
+    if (!promotionName) {
+      return;
+    }
+    promotionName = promotionName.getAttribute('content').toLowerCase();
+
+    try {
+      const { locale } = window.fedPub;
+      const promotion = await window.fetch(`${locale === 'en' ? '' : `/${locale}`}/promotions/hub/${promotionName}.plain.html`);
+      const promotionContent = await promotion.text();
```

Comment:
Why do you need this try/catch ? Feels like you are uncertain of the code behaviour.
I think the only thing that could go wrong is if the `fetch` fails which is something you could catch before trying to get the `text()` and by testing `promotion.ok`.